powertools-parameters/src/main/java/software/amazon/lambda/powertools/parameters/DynamoDBProvider.java

```diff
@@ -0,0 +1,74 @@
+package software.amazon.lambda.powertools.parameters;
+
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.*;
+import software.amazon.lambda.powertools.parameters.cache.CacheManager;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class DynamoDBProvider extends BaseProvider {
+
+    private final DynamoDbClient client;
+    private final String tableName;
+
+    public DynamoDBProvider(CacheManager cacheManager, String tableName) {
+        this(cacheManager, DynamoDbClient.builder()
+                .httpClientBuilder(UrlConnectionHttpClient.builder())
```

Comment:
@jeromevdl , is there any reason that this code in e.g. the SecretsProvider explicitly configures the AWS SDK client? My experience has been that the default configuration chain is usually a safe bet - am I missing something?

https://github.com/awslabs/aws-lambda-powertools-java/blob/b062a7622d6845e1ea775a7dcd7c3658315e6dfa/powertools-parameters/src/main/java/software/amazon/lambda/powertools/parameters/SecretsProvider.java#L66-L72

Reply:
Actually I don't know/remember. But it works, `AWS_ACCESS_KEY`, `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN` are available: https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime

Reply:
For the http client, we could now move to CRT (https://aws.amazon.com/blogs/developer/announcing-availability-of-the-aws-crt-http-client-in-the-aws-sdk-for-java-2-x/).

Reply:
`.credentialsProvider(EnvironmentVariableCredentialsProvider.create())` was definitely the best way to configure clients pre SnapStart. Now it's probably best to use the default.

Reply:
Yep, I remember now. By specifying environment variables, the SDK goes directly that way and doesn't look for multiple sources. It's faster. Not sure everyone is using snap start though. Unless it is a problem for snap start (is it?), I would keep it like that.

Reply:
Thanks both! I'll plug in CRT and use the env vars. It's nice if it generalizes outside of Lambda, but our explicit goal here is to run in Lambda, so it's a bit unnecessary.